### ISSUE TITLE
How to Fast - Xgboost Hyperparameter Search

### ISSUE THREAD

**Issue by avinashbarnwal** [avinashbarnwal/aftXgboostPaper#4]
Hi @hcho3 and Prof. @tdhock 

We have 18000 combinations for hyperparameter tuning. Please find this code. 
Link - https://github.com/avinashbarnwal/aftXgboostPaper/blob/master/src/R/production/xgboost/xgboost_hyper.ipynb

I am looking to optimize this. Please let me know if you have any ideas to make it fast.

**Comment by tdhock** [comment]
I am not an expert on xgboost hyper-parameters. I thought you already discussed this with @hcho?

**Comment by avinashbarnwal** [comment]
We discussed but didn't realize it would take this much time in R.

**Comment by hcho** [comment]
Please double check who you are mentioning.

**Comment by chyunsu3** [comment]
@avinashbarnwal Can you be more specific? How long is each combination taking?

**Comment by avinashbarnwal** [comment]
Hi @hcho3,

It is taking ~42 secs for running one iteration.

Please find the code here for one iteration - https://github.com/avinashbarnwal/aftXgboostPaper/blob/master/src/R/production/xgboost/xgboost_hyper.ipynb

**Comment by avinashbarnwal** [comment]
Hi @hcho3, 
As we discussed, I am porting the code from R to python for hyper-parameter tuning. As grid-search is very slow in R and support for packages like optuna is not there for R.

https://github.com/pfnet/optuna

**Comment by chyunsu3** [comment]
I also granted @avinashbarnwal access to a fast machine.

**Comment by avinashbarnwal** [comment]
Thanks @hcho3.

**Comment by avinashbarnwal** [comment]
Hi Prof. @tdhock and @hcho3 ,

I have the results for intervalCV , survival regression and xgboost.

https://github.com/avinashbarnwal/aftXgboostPaper/tree/master/result/ATAC_JV_adipose

Please let me know your thoughts.

**Comment by tdhock** [comment]
did you double-check the predictions/accuracy of IntervalRegressionCV using my precomputed files, as mentioned in #2 ?

**Comment by tdhock** [comment]
also did you run it for all data sets or just the ATAC data set?

**Comment by tdhock** [comment]
also are the 1.csv 2.csv etc files predictions? if so they should have a column for sequenceID so we can compute accuracy metrics. please add one.

**Comment by avinashbarnwal** [comment]
> also did you run it for all data sets or just the ATAC data set?

I have done it only for ATAC data set.

**Comment by avinashbarnwal** [comment]
> also are the 1.csv 2.csv etc files predictions? if so they should have a column for sequenceID so we can compute accuracy metrics. please add one.

Done. Check the particular file - https://github.com/avinashbarnwal/aftXgboostPaper/blob/master/result/ATAC_JV_adipose/intervalCV/%201%20.csv

**Comment by avinashbarnwal** [comment]
> did you double-check the predictions/accuracy of IntervalRegressionCV using my precomputed files, as mentioned in #2 ?

I don't think this is test fold rather cross-validation folders. Please check on the last of the notebook. - https://github.com/avinashbarnwal/aftXgboostPaper/blob/master/src/R/production/penaltyLearning/intervalCV.ipynb

I think we need to test the test-folds, not the cross-validation folds.

**Comment by tdhock** [comment]
good that you store sequenceIDs in prediction files now.

also what distribution did you use? I think you should compute predictions files for all distributions, and for all test folds.

for my predictions.csv files there is one row for each sequenceID in the test set. Yours should be too. It looks like your predictions are for different sequenceIDs. You should double-check your code.

**Comment by avinashbarnwal** [comment]
This is based on the best results of cross-validation folds. Now I am considering distributions as hyper-parameter. 

I am making the folds based on the folds data folder. I would like to have a quick call to weed out this hiccup.
